caps/models.py

```diff
@@ -110,6 +110,58 @@ def __str__(self):
     def get_absolute_url(self):
         return "/councils/%s/" % self.slug
 
+    def get_related_councils(self, cut_off: int = 10) -> List[dict]:
+        """
+        get all related councils
+        """
+        councils = (self
+                    .related_authorities
+                    .filter(distances__position__lte=cut_off + 2)
+                    .annotate(num_plans=Count('plandocument'),
+                              has_promise=Count('promise'),
+                              earliest_promise=Min('promise__target_year'),
+                              declared_emergency=Min('emergencydeclaration__date_declared')))
+
+        council_lookup = {x.id: x for x in councils}
+        council_ids = list(council_lookup.keys()) + [self.id]
+
+        # get label assignments in a seperate query ready to assign
+        labels = (ComparisonLabelAssignment
+                  .objects
+                  .filter(council_id__in=council_ids)
+                  .prefetch_related("label")
+                  .order_by("label__type_id"))
+
+        # change list of labels
+        grouped_labels = groupby(labels, lambda x: x.label.type_id)
+
+        # change list of labels to tiered lookup (type_id, council_id)
+        labels = {k: {y.council_id: y for y in v} for k, v in grouped_labels}
+
+        processed_councils = []
+        for d in self.distances.all():
+            nc = deepcopy(council_lookup[d.council_b_id])
```

Comment:
This is making it create a new object that avoids any of its properties being references that might be shared between multiple instances. 

The situation I want is that the same council may appear in multiple list of comparisons depending on the comparison type.

This *might* not be necessary, but I think there's a risk that `processed_councils` may end up containing multiple councils that are just references to the same object.

What I'm trying to avoid is a situation where the next line is assigning a distance to the council object, but the same council may come up in a separate comparison, be referenced again, and have a new distance assigned to it.

Reply:
Added